keel-core/src/main/kotlin/com/netflix/spinnaker/keel/services/ApplicationService.kt

```diff
@@ -165,10 +171,14 @@ class ApplicationService(
       environment,
       status.type
     )
```

Comment:
The way this is, you'll only notify plugins if a user changed the constraint state. We publish constraint state changed events in a couple of places, because sometimes the system changes the state.

I think it might be better to create a new component that listens to all `ConstraintStateChanged` events and then  notifies each stateful evaluator. That way you only have to add this logic in one place.

Reply:
Yup, good call! I had the same exact thought the minute after I pushed this. Will make that change!

Reply:
Updated the PR based on your suggestion.